src/msgraph_core/tasks/large_file_upload.py

```diff
@@ -144,7 +119,15 @@ async def upload(self, after_chunk_upload: Optional[Callable] = None):
                 logging.error("Error uploading chunk  %s", error)
             finally:
                 self.chunks -= 1
-        return session
+        upload_result = UploadResult()
+        upload_result.upload_session = UploadSessionDataHolder(
+            expiration_date_time=self.upload_session.expiration_date_time,
+            next_expected_ranges=self.upload_session.next_expected_ranges,
+            upload_url=self.upload_session.upload_url
+        )
+        upload_result.item_response = session
```

Comment:
The final chunk upload doesn't return an upload session but returns an [empty body with `Location` header](https://learn.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http#step-3-continue-uploading-byte-ranges-until-the-entire-file-has-been-uploaded) for Outlook & a [`DriveItem` object](https://learn.microsoft.com/en-us/graph/api/driveitem-createuploadsession?view=graph-rest-1.0#completing-a-file) for OneDrive/Sharepoint. 

The last chunk upload should deserialize the response to a ParsableFactory that is passed perhaps via the [constructor](https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/dev/src/main/java/com/microsoft/graph/core/tasks/LargeFileUploadTask.java#L66). This is a gap in PHP as well. So that the developer can pass the type to deserialize the final response to e.g. [`DriveItem::createFromDiscriminatorValue` or `AttachmentItem` ](https://learn.microsoft.com/en-us/graph/sdks/large-file-upload?tabs=csharp)